sem_covid/adapters/embedding_models.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+# Date    : 20.07.2021 
+# Author  : Stratulat Ștefan
+# File    : embedding_models.py
+# Software: PyCharm
+from typing import List
+
+import numpy as np
+from gensim.models import KeyedVectors
+
+from sem_covid.adapters.abstract_model import (WordEmbeddingModelABC, SentenceEmbeddingModelABC,
+                                               TokenizerModelABC)
+import tensorflow_hub as hub
+
+import pandas as pd
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+from transformers import AutoTokenizer, AutoModel, BertModel, BertTokenizerFast
+
+
+class BasicTokenizerModel(TokenizerModelABC):
+
+    def tokenize(self, text: str) -> List[str]:
+        return text.split(' ')
+
+
+class SpacyTokenizerModel(TokenizerModelABC):
+
+    def __init__(self, spacy_nlp):
+        self._spacy_nlp = spacy_nlp
+
+    def tokenize(self, text: str) -> List[str]:
+        return list(map(str, self._spacy_nlp(text)))
+
+
+class Word2VecEmbeddingModel(WordEmbeddingModelABC):
+
+    def __init__(self, word2vec: KeyedVectors):
+        self._word2vec = word2vec
+        self._vector_dimension = word2vec.vector_size
+
+    def encode(self, tokens: List[str]) -> List:
+        return [self._word2vec[word].tolist() if word in self._word2vec else np.zeros(self._vector_dimension).tolist()
+                for word in tokens]
+
+
+class AverageSentenceEmbeddingModel(SentenceEmbeddingModelABC):
+
+    def __init__(self, word_embedding_model: WordEmbeddingModelABC, tokenizer: TokenizerModelABC):
+        self._word_embedding_model = word_embedding_model
+        self._tokenizer = tokenizer
+
+    def encode(self, sentences: List[str]) -> List:
+        results = []
+        for sentence in sentences:
+            embeddings = self._word_embedding_model.encode(self._tokenizer.tokenize(sentence))
+            results += [np.mean([np.zeros(len(embeddings[0])).tolist()] + embeddings, axis=0).tolist()]
+        return results
+
+
+class TfIdfSentenceEmbeddingModel(AverageSentenceEmbeddingModel):
+
```

Comment:
lets document this class, what are the intentions and how it should function;
and how it should be used :)